### PR TITLE
[Java] Remove RayRuntimeInternal class

### PR DESCRIPTION
**Pull request by jovany-wang**
## Why are these changes needed?
Due to we have already removed the multiple workers in one process, remove RayRuntimeInternal for purpose.

## Related issue number
#24567

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: java/runtime/src/main/java/io/ray/runtime/RayDevRuntime.java [hunk 1]
```diff
@@ -33,7 +33,6 @@ public RayDevRuntime(RayConfig rayConfig) {
     super(rayConfig);
   }
 
-  @Override
```
Comment (kfstorm) [inline java/runtime/src/main/java/io/ray/runtime/RayDevRuntime.java:36]: Why remove these in `RayDevRuntime`? I believe the methods are defined in `AbstractRayRuntime`.

**Review by kfstorm — APPROVED** [review]

File: java/runtime/src/main/java/io/ray/runtime/RayDevRuntime.java [hunk 2]
```diff
@@ -57,7 +56,6 @@ public void start() {
             });
   }
```
Comment (jovany-wang) [inline java/runtime/src/main/java/io/ray/runtime/RayDevRuntime.java:58]:
```suggestion
@Override
```

**Comment by jovany-wang** [comment]
This only changes Java part. Failure tests are not related.

**Comment by jovany-wang** [comment]
This only changes Java part. Failure tests are not related.

**Pull request merged**